### PR TITLE
lint: upgraded to 1.22.2 and make lint issues a build failure

### PR DESCRIPTION
**Pull request by jkowalski**
fixed or silenced linter warnings, mostly due to magic numeric constants

### CODE REVIEW

File: tools/tools.mk [hunk 1]
```diff
@@ -35,11 +35,12 @@ $(BINDATA_TOOL):
 	go build -o $(BINDATA_TOOL) github.com/go-bindata/go-bindata/go-bindata
 
 # linter
-LINTER_TOOL=$(TOOLS_DIR)/bin/golangci-lint
+LINTER_TOOL=$(TOOLS_DIR)/bin/golangci-lint-$(GOLANGCI_LINT_VERSION)
```
Comment (julio-lopez) [inline tools/tools.mk]: nice!

File: repo/content/index.go [hunk 2]
```diff
@@ -164,10 +164,6 @@ func (b *index) GetInfo(contentID ID) (*Info, error) {
 }
 
 func (b *index) entryToInfo(contentID ID, entryData []byte) (Info, error) {
-	if len(entryData) < 20 {
```
Comment (julio-lopez) [inline repo/content/index.go]: Is this check performed inside `e.parse` ?

File: internal/fshasher/fshasher.go [hunk 3]
```diff
@@ -114,9 +114,10 @@ func writeFile(ctx context.Context, w io.Writer, f fs.File) error {
...
 	}
-	defer r.Close()
 
-	_, err = io.Copy(w, r)
+	if _, err = io.Copy(w, r); err != nil {
+		return err
+	}
 
-	return err
+	return r.Close()
```
Comment (julio-lopez) [inline internal/fshasher/fshasher.go]: This actually changes the behavior and would leave `r` open when `io.Copy` fails.

File: repo/content/index.go (hunk 2 above)
Comment (jkowalski) [reply repo/content/index.go]: Yes

File: internal/fshasher/fshasher.go (hunk 3 above)
Comment (jkowalski) [reply internal/fshasher/fshasher.go]: fixed

Comment (jkowalski) [reply internal/fshasher/fshasher.go]: one sec, let me fix it again :)

Comment (jkowalski) [reply internal/fshasher/fshasher.go]: done, I initially misread the code and thought that it's the `w` that needs to be closed, simply adding `//nolint:errcheck` is the right thing to do for read-only streams

**Pull request merged**